crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -104,18 +104,16 @@ impl KvRedisProvider {
 
     #[instrument(level = "debug", skip(self))]
     async fn invocation_conn(&self, context: Option<Context>) -> anyhow::Result<ConnectionManager> {
-        let Some(Context {
-            component: Some(source_id),
-            tracing,
-        }) = context
-        else {
+        let ctx = context.context("unexpectedly missing context")?;
+
```

Comment:
When would this happen, could we add any extra information here?

Reply:
Well this is just a result of the contract as it's written -- the custom interface libraries define the fucntions with an optional context:

```rust
impl keyvalue::store::Handler<Option<Context>> for KvRedisProvider {
    #[instrument(level = "debug", skip(self))]
    async fn delete(
        &self,
        context: Option<Context>,
        bucket: String,
        key: String,
    ) -> anyhow::Result<Result<()>> {
        propagate_trace_for_ctx!(context);
        check_bucket_name(&bucket);
        Ok(self.exec_cmd(context, &mut Cmd::del(key)).await)
    }
```

So this bleeds through to `exec_cmd` and eventually `invocation_conn`... There's no good reason Context would be missing (and it shouldn't ever be), so there's not a ton more to say other than that...